staff/jordi-pujol/horroy/scripts/horroy.js

```diff
@@ -0,0 +1,286 @@
+function Horroy() {
+
+    if ((arguments.length === 1) && (!(Number.isInteger(arguments[0]))) && (typeof arguments[0] === "number")) {
+        throw Error('Invalid horroy length')
+        return undefined
+    }
+
+    else if ((arguments.length === 1) && (Number.isInteger(arguments[0]))) {
+
+        this.length = arguments[0]
+
+        for (var i = 0; i < arguments[0]; i++) {
+            this[i] = undefined
+        }
+    }
+
+    else if ((arguments.length > 1) || ((arguments.length === 1) && (typeof arguments[0] !== "number"))) {
+
+        this.length = arguments.length
+
+        for (var i = 0; i < arguments.length; i++) {
+            this[i] = arguments[i]
+        }
+    }
+}
+
+Horroy.prototype.push = function (value) {
+
+    for (var i = 0; i < arguments.length; i++) {
+
+        // if((typeof arguments[i] !== "number") && (!(arguments[i] instanceof Function)) && (!(arguments[i] instanceof Object)) && (!(arguments[i] instanceof Horroy)) && (typeof arguments[i] !== "undefined") && (!(typeof arguments[i] !== null))) throw TypeError ('not recognized param')
+
+        this[this.length] = arguments[i];
+        this.length++
+    }
```

Comment:
you could also write `this[this.length++] = arguments[i];`
In this way you save a line (But if you do it in all the functions you can save quite a lot)
In this way, when the `++` are after `this.length`. The current number is executed and after it is sum 1

Reply:
Thanks! you're right, I'll change it :)